### ISSUE TITLE
Subrepo Organization

### ISSUE THREAD

**Issue by FranticRain** [FranticRain/HONKS#1]
**Context**

One of the big issues with QMK is that the keymaps are managed inside of the main repository. For HONKS, I propose an alternate structure. There will be a main firmware repository, let's call it `firm`. And then we have a `keymap` repository that will store all standard keymaps. `firm` will include `keymap` as a sub-repo. The `keymap` repo will consist of a collection of submodules to other repositories where the specifics of keyboards are implemented. This means that when you clone the repository, you can clone only the submodule that you need for the keyboards that you want. And if you implement alternative repositories for your keyboards, then you are responsible for rebasing your repository for that keyboard.

This should help reduce bloat and keep our repositories organized and focused.

**Alternatives**

Alternative would likely be QMK-style, but responses are open to other options.

**Has the feature been requested before?**

N/A

**If the feature request is approved, would you be willing to submit a PR?**

Yes

**Comment by FrancisUsher** [comment]
Seems like a good idea. One advantage the QMK method has is that since it's a monorepo, all the code in one version is ostensibly compatible. To make up for this it probably makes sense to decide on a versioning system to be able to infer compatibility between the subrepos.

**Comment by FranticRain** [comment]
Yeah, that's a good idea. Perhaps part of your configuration file (be it JSON, YAML, etc) should probably be assigning a HONKS firmware target version. Keymaps should probably similarly have a keyboard and firmware version attached to them. That way, when you pull a keymap, we can target the correct version of everything for successful compilation, and provide warnings that you are behind master/latest.
